docs/doc-system.md

```diff
@@ -0,0 +1,114 @@
+Slang Doc System 
+================
+
+Slang contains a rudimentary documentation generation system. The mechanism used to mark up source is similar to [doxygen](https://www.doxygen.nl/manual/docblocks.html). Namely
+
+```
+/**
+ ... text ... (JavaDoc style)
+ */
+void someFunctionA() {} 
+ 
+/*!
+ .. text .. (QT style)
+ another line
+ */ 
+void someFunctionB() {} 
+ 
+/// ... text ... (Multi line)
+/// another line
+void someFunctionC() {}
+ 
+//!... text ...  (QT Multi line)
+//! another line
+void someFunctionD() {}
+ 
+```
+ 
+All of the above examples will add the documentation for the declaration that appears after them. Also note that this slightly diverges from doxygen in that an empty line before and after in a multi line comment is *not* required.
```

Comment:
Can we have a "comparison with doxygen" section and move all the detailed differences there? The way this is phrased here kind of ends up assuming that the reader is already familiar with doxygen, and anticipates that they will already be thinking of detailed questions like that.

A reader who doesn't already know doxygen's conventions might just be confused by this bit, since all of the above being equivalent seems like the obvious behavior anyway.